### PR TITLE
RELATED: ONE-3056 Bubble into production

### PR DESCRIPTION
**Pull request by peterlacko**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/219/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1457/ : FAILURE in 1m 02s

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/300/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1537/ : FAILURE in 1m 43s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/305/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1542/ : SUCCESS in 4m 13s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/324/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1558/ : SUCCESS in 4m 20s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/328/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1562/ : SUCCESS in 1m 59s

**Comment by peterlacko** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/637/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/331/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/524/ : UNSTABLE in 15m 47s

**Comment by peterlacko** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/336/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1572/ : SUCCESS in 4m 35s

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/341/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1576/ : SUCCESS in 1m 47s

File: src/components/afm/BubbleChart.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export {
+    IDataSourceProviderProps
+};
+
+import { ICommonChartProps } from '../core/base/BaseChart';
+import { BubbleChart as CoreBubbleChart } from '../core/BubbleChart';
+import { generateDefaultDimensionsForPointsCharts } from '../../helpers/dimensions';
+
+/**
+ * AFM ScatterPlot
```
Comment (ZdenekVidensky) [inline src/components/afm/BubbleChart.tsx]: Change to AFM BubbleChart

File: src/components/afm/BubbleChart.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * is an internal component that accepts afm, resultSpec
+ * @internal
+ */
+export const BubbleChart = dataSourceProvider<ICommonChartProps>(
+    CoreBubbleChart,
+    generateDefaultDimensionsForPointsCharts,
+    'ScatterPlot'
```
Comment (ZdenekVidensky) [inline src/components/afm/BubbleChart.tsx]: BubbleChart

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: examples/test/BasicComponents_test.js [hunk 3]
```diff
@@ -94,3 +94,12 @@ test('Scatter plot should render', async (t) => {
         .expect(chart.exists).ok()
         .expect(chart.textContent);
 });
+
+test('Bubble chart should render', async (t) => {
+    const loading = Selector('.s-loading');
```
Comment (pbenes) [inline examples/test/BasicComponents_test.js]: what would you think of having a helper like (not tested):
```
async function checkRenderChart(selector) {
    const loading = Selector('.s-loading');
    const chart = Selector(selecor);
    await t
        .expect(loading.exists).ok()
        .expect(chart.exists).ok()
        .expect(chart.textContent);
}
```
and then just call from those tests above (just those which do exactly this)
```
test('Bubble chart should render', async (t) => {
    await checkRenderChart('.s-bubble-chart')
}
```

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/346/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1581/ : SUCCESS in 2m 17s

File: src/components/visualizations/chart/chartOptionsBuilder.ts [hunk 4]
```diff
@@ -365,18 +367,32 @@ export function getScatterPlotSeries(
...
+    const primaryMeasuresBucket = get(mdObject, ['buckets'], [])
+        .find(bucket => bucket.localIdentifier === 'measures');
+    const secondaryMeasuresBucket = get(mdObject, ['buckets'], [])
+        .find(bucket => bucket.localIdentifier === 'secondary_measures');
+
+    const primaryMeasuresBucketEmpty = isEmpty(get(primaryMeasuresBucket, 'items', []));
+    const secondaryMeasuresBucketEmpty = isEmpty(get(secondaryMeasuresBucket, 'items', []));
+
     return executionResultData.map((resData: any, index: number) => {
+        let data: any = [];
```
Comment (pbenes) [inline src/components/visualizations/chart/chartOptionsBuilder.ts]: Seems to me like pretty complicated logic. Do we have any test for that?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/348/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1582/ : SUCCESS in 5m 16s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/349/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1584/ : SUCCESS in 3m 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/350/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1585/ : SUCCESS in 3m 38s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/351/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1586/ : SUCCESS in 1m 46s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/352/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1587/ : SUCCESS in 1m 36s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/353/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1588/ : SUCCESS in 7m 19s

**Comment by peterlacko** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/354/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1589/ : SUCCESS in 4m 02s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/644/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/355/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/527/ : SUCCESS in 18m 14s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/57/ : SUCCESS in 18s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1591/ : SUCCESS in 1m 41s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/645/ : SUCCESS in 17m 06s

**Pull request merged**